### PR TITLE
Re-add uaccess tag for /dev/kvm

### PR DESCRIPTION
**Pull request by mbiebl**
If --dev-kvm-mode is set to something different then 0666, which we
explicitly support, it makes sense to still apply the uaccess tag to
/dev/kvm. For distros which opt to use the default 0666, this change is
a nop.

This partially reverts commit b8fd3d82205f632ce001fade74fed287e1564a1a.

### CODE REVIEW

**Comment by poettering** [comment]
I#d rather not do uaccess if the access mode is 0666 anyway. Can you conditionalize that?

Also, why do this for kvm only, not for the other stuff?

**Comment by AlexanderKurtz** [comment]
Hey, @mbiebl and @poettering, I have been silently following this since the Debian BTS notified me that my bug 887852 [0] was marked ```pending```. After some poking around, I found that it was probably bug 892945 [1] which made this problem resurface again. Reading through that bug report, I found this statement from Ben Hutchings (one of Debian's Linux maintainers, possibly @bwhacks) [2]:

> It is fairly mature, but it still has a large attack surface and
occasional security issues that can be exploited by the VM owner.  So I
think it make sense to restrict access to the kvm group and local
logins.  This should mitigate the security issues on multiuser systems
without too much disruption.

My question now is this: How can the permissions of ```/dev/kvm``` be a distro-specific setting? Aren't all distros ultimately affected by such potential vulnerabilities in the KVM subsystem and aren't local privilege escalation attacks equally as bad if the happen on Fedora as if they happen on Debian? I think that one of the most important accomplishments of systemd is a decent amount of standardization between the Linux distros, and I think systemd should make a clear recommendation to the distros regarding the permissions of ```/dev/kvm```.

[0] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=887852
[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892945
[2] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892945#62

**Comment by mbiebl** [comment]
> I#d rather not do uaccess if the access mode is 0666 anyway. Can you conditionalize that?

I can update the patch if you want that. This most likely means turning ./src/login/70-uaccess.rules into a .m4 processes file, if that is ok.

Btw, what's the point of making the device root:kvm if it's 0666? That seems odd.

> Also, why do this for kvm only, not for the other stuff?

Not sure what you mean by that. We apply the uaccess tag to a lot of devices.

**Comment by mbiebl** [comment]
@poettering updated as per your request

Since this touches the build system, @keszybz 's review would be most welcome

File: meson.build [hunk 1]
```diff
@@ -741,7 +741,9 @@ substs.set('USERS_GID', users_gid)
 conf.set10('ENABLE_ADM_GROUP', get_option('adm-group'))
 conf.set10('ENABLE_WHEEL_GROUP', get_option('wheel-group'))
 
-substs.set('DEV_KVM_MODE', get_option('dev-kvm-mode'))
+dev_kvm_mode = get_option('dev-kvm-mode')
+substs.set('DEV_KVM_MODE', dev_kvm_mode)
+conf.set10('DEV_KVM_UACCESS', dev_kvm_mode != '0666')
```
Comment (filbranden) [inline meson.build:746]: nitpick: Does an integer comparison work here?

```python
conf.set10('DEV_KVM_UACCESS', int(dev_kvm_mode) != 0o0666)
```

And should this maybe check whether the "other" bits are set, regardless of which of them are?

```python
conf.set10('DEV_KVM_UACCESS', (int(dev_kvm_mode) & 0o7) != 0)
```

(This last one probably needs a comment to clarify what that does...)

Comment (mbiebl) [reply meson.build:746]: Thanks for your review.
What would be the benefit of doing an integer comparison? Tbh, I find the version with the string comparison a bit more readable. But I can change it if requested.

**Comment by mbiebl** [comment]
@poettering I've made the changes you requested. PTAL

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**